### DISCUSSION TITLE
[Tour Editor - Weather Text] Keeping the user's weather input

### DISCUSSION THREAD

**Discussion by man-pro** [mytourbook/mytourbook#1069] [Ideas]
The textfield _Weather_ in the _Tour Editor_ is always filled with weather data. Unfortunately, this also deletes all your own entries. If you want to call the historical data, at first you have to copy your own content and after you have to paste it into the field back. This may be ok for a few tours, but ... It would be great if the contents of the field were preserved.

Maybe you can "mark" your entries. Something like this. This way you could also enter the data of different providers.

> Sunny, warm day, a bit windy
> 
> --- mt start / do not edit this section ---
> Retrieving weather data with "WorldWeatherOnline" 13/04/2023, 11:01:39
> 25/09/2021, 11:11: ☀ Sunny, 22°C, max. 24°C, min. 20°C, feels like 23°C, 6km/h from S, 50% humidity, 1016.71mbar pressure
> 11h 20°C feels like 20°C 6km/h from 193° humidity 59% pressure 1018.0mbar precipitation 0.0mm snowfall 0.0mm
> 12h 22°C feels like 22°C 6km/h from 193° humidity 53% pressure 1017.0mbar precipitation 0.0mm snowfall 0.0mm
> 13h 22°C feels like 25°C 6km/h from 192° humidity 50% pressure 1017.0mbar precipitation 0.0mm snowfall 0.0mm
> 14h 24°C feels like 25°C 6km/h from 191° humidity 47% pressure 1017.0mbar precipitation 0.0mm snowfall 0.0mm
> 15h 24°C feels like 25°C 6km/h from 191° humidity 44% pressure 1016.0mbar precipitation 0.0mm snowfall 0.0mm
> 16h 23°C feels like 24°C 5km/h from 182° humidity 47% pressure 1016.0mbar precipitation 0.0mm snowfall 0.0mm
> 17h 21°C feels like 21°C 4km/h from 174° humidity 50% pressure 1016.0mbar precipitation 0.0mm snowfall 0.0mm
> 
> Weather Station: "Friedberg", Distance from tour: 4.0km
> --- mt end ---
>

**Comment by wolfgang-ch** [comment]
@FJBDev A simple solution could be, not setting the weather but append the weather from the weather retrieval to the existing weather text, what do you think?

**Comment by wolfgang-ch** [comment]
Implemented in 23.10 https://github.com/mytourbook/mytourbook/issues/1191

**Reply by FJBDev** [reply]
I don't like this solution as it creates potentially issues when retrieving the weather twice:

![image](https://github.com/mytourbook/mytourbook/assets/25235665/b268cf89-bfe8-44ff-b136-af88d8f66bc6)

**Comment by wolfgang-ch** [comment]
I made it optional to append the weather description

![image](https://github.com/mytourbook/mytourbook/assets/1283445/ca081182-38a8-4a9f-80e4-74b3d9c6ebe2)

The tooltip do now show more info what would happen when pressing the weather link

![image](https://github.com/mytourbook/mytourbook/assets/1283445/c12ed8f8-9594-4444-b86e-d9a056f0dc52)

**Reply by wolfgang-ch** [reply]
@FJBDev Can you test this new feature if it works and can make you happy 😄

**Reply by FJBDev** [reply]
@wolfgang-ch  Feature tested and it looks good 👍
